R/init.R

```diff
@@ -153,6 +153,21 @@ init <- function(data,
   # log
   teal.logger::log_system_info()
 
+  # invoke lockfile creation
+  # If user has setup the file, there is no need to compute a new one.
+  user_lockfile <- getOption("teal.renv.lockfile", "")
+  if (!(file.exists(user_lockfile) || is_in_test() || is_r_cmd_check())) {
+    old_plan <- future::plan()
+    # If there is already a parallel backend, reuse it.
+    if (inherits(old_plan, "sequential")) {
+      future::plan(future::multisession, workers = 2)
+    }
+
+    lockfile_task <- ExtendedTask$new(create_renv_lockfile)
+    lockfile_task$invoke(inherits(old_plan, "sequential"))
+    logger::log_trace("lockfile creation invoked.")
+  }
```

Comment:
let's move this into a single function and have a one-liner in init()

Reply:
if moved - you might consider creating a new file in `R/` and put there also `create_renv_lockfile`.
Feel free to do renames if needed

Reply:
moved in here 1d6ef3005c508c3c6b7963739238ff08a38db195